### ISSUE TITLE
frep-loop: Nested loops not treated correctly

### ISSUE THREAD

**Issue by noah95** [pulp-platform/snitch-llvm#5]
A nested loop with frep inference in the inner loop is not detected in the FREP loop pass.

Failing code:
```c
void frep_not_ok(double *out, unsigned K) {
  register double acc = 0.0;

  __builtin_ssr_enable();
  for (unsigned j = 0; j < K; j ++) {
    #pragma frep infer
    for (unsigned j = 0; j < 1024; j ++) {
      acc += __builtin_ssr_pop(0)*__builtin_ssr_pop(1);
    }
    out[j] = acc;
  }
  __builtin_ssr_disable();
}
```

Output
```c
Loop at depth 1 containing: 
 %bb.3<header>,%bb.4,%bb.6,%bb.7,%bb.8,%bb.5,%bb.10,%bb.11<latch><exiting>
Loop at depth 2 containing: 
 %bb.6<header>,%bb.7,%bb.8<latch><exiting>
Loop at depth 3 containing: 
 %bb.7<header><latch><exiting>
Loop at depth 2 containing: 
 %bb.10<header><latch><exiting>
Nested loop not supported
```

**Comment by noah95** [comment]
Added support for this in #10
